plutus-core/plutus-ir/src/PlutusIR/Transform/RewriteRules.hs

```diff
@@ -0,0 +1,59 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs             #-}
+{-# LANGUAGE RankNTypes        #-}
+{-# LANGUAGE TypeOperators     #-}
+module PlutusIR.Transform.RewriteRules
+    ( userRewrite
+    , RewriteRules (..)
+    , defaultUniRewriteRules
+    ) where
+
+import PlutusCore.Default
+import PlutusCore.Name
+import PlutusCore.Quote
+import PlutusIR as PIR
+import PlutusIR.Analysis.VarInfo
+import PlutusIR.Transform.RewriteRules.CommuteFnWithConst
+import PlutusPrelude
+
+import Control.Lens
+
+
+-- | Rewrite a `Term` using the given `RewriteRules` (similar to functions of Term -> Term)
+-- Normally the rewrite rules are configured at entrypoint time of the compiler.
+userRewrite :: ( Semigroup a, t ~ Term tyname name uni fun a
```

Comment:
How does a user specify a rewrite rule? It doesn't seem possible.

Reply:
userRewrite is a misnomer. What I wanted to express is that this `userRewrite` function is a general transformation that accepts "configurable" rewrite rules, based on a specific fun/universe. 

Currently our tx->pir->uplc pipeline fixes fun/uni to be `DefaultFun`, `DefaultUni`. The "user" does not have the ability to specify custom rules (like RULES of GHC), but he/she could statically configure another list of rewrite rules.

Anyway a misnomer, I will change it to sth like `applyRewriteRules`